src/lib/components/table/components/TableSection/TableSection.tsx

```diff
@@ -97,40 +102,28 @@ const TableSection = ({
           return (
             rowIndex > 0 && (
               <tr key={rowIndex} className={styles.tr}>
-                {row.map((tableCellProps, cellIndex) => {
+                {row.map((tableCellData, cellIndex) => {
                   const key = `${rowIndex}-${cellIndex}`;
                   const isFirstCellInRow = cellIndex === 0;
 
-                  const onCelInfoClick = (body: ReactNode) =>
-                    handleOpenModal({
-                      cellIndex,
-                      body,
-                      title: isFirstCellInRow
-                        ? tableCellProps.content
-                        : undefined,
-                    });
-
-                  let openModal;
-                  let align: Alignment = 'left';
-
-                  const isBaseCell = !tableCellProps.type;
-
-                  if (isBaseCell) {
-                    openModal = onCelInfoClick;
-                    align = isFirstCellInRow ? 'left' : 'center';
-                  }
-
                   return (
                     <TableCell
-                      {...(isBaseCell
+                      isFirstCellInRow={isFirstCellInRow}
+                      key={key}
+                      {...tableCellData}
+                      {...(isBaseCell(tableCellData)
                         ? {
-                            openModal,
-                            align,
+                            openModal: (body: ReactNode) =>
+                              openModal?.({
+                                body,
+                                title: isFirstCellInRow
+                                  ? getModalTitleFromRowHeader(row)
+                                  : getModalTitleFromColumnHeader(cellIndex) ||
+                                    getModalTitleFromRowHeader(row),
```

Comment:
wdyt about moving this to a separate function/constant just for improving readibility?

Reply:
Sure!